### PR TITLE
Update pyjwt required version; remove py2 support

### PR DESCRIPTION
**Pull request by pletnes**

### CODE REVIEW

**Comment by pletnes** [comment]
@robertofalk I discovered multiple issues related to python 2 compatibility in the repository. I have tried to cleanup and fix these in this PR, so we can move forward with development. I believe python 2 support is no longer relevant since it has been end-of-life since January 1, 2020.

**Comment by robertofalk** [comment]
I think it look good, but before merging it I'll need to check if SAP has any official statement regarding dropping python2.7 support, and how this can be done here. Also it's safer to create a branch where we keep supporting python2.7, so in case of security fixes, we can still apply in that branch. I'll get back to you on this.

**Comment by pletnes** [comment]
@robertofalk the travis/python3.6 build failed due to an obsolete virtualenv version. A `pip install -U virtualenv` fixed the build. Now, 3.6, 3.7, 3.8 are working. I am also able to run `tox` tests inside the Docker image, as before, although python2 is now removed from the image.

Are we ready to merge now, or do you have further concerns?

**Comment by pletnes** [comment]
Note that build 193 passes, but that the latest builds seem to no longer be scheduled. Are we running up against a limitation of the Travis free tier?

File: README.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ change the `USE_SAP_PY_JWT` environment variable to `true`.
 
 
 # Requirements
-*sap_xssec* requires either *python 2.7* or *python 3.7*.
+*sap_xssec* requires either *python 3.7* or newer.
```
Comment (robertofalk) [inline README.md:121]: @pletnes 3.6, since you fixed the issue with travis!

**Comment by robertofalk** [comment]
Besides the fix, I think it looks good.

**Comment by pletnes** [comment]
Great attention to detail, @robertofalk! I've fixed it, is it better now?

**Comment by robertofalk** [comment]
@pletnes I think it only makes sense to merge it if the code replacing requests with httpx is ready and tested. Either we go 100% for httpx or we stay as it is

**Comment by robertofalk** [comment]
Let me add a message to the README explaining the drop of python 2.7 support, then we can merge.

**Review by robertofalk — APPROVED** [review]

**Pull request merged**